baseapp/abci.go

```diff
@@ -758,7 +758,11 @@ func (app *BaseApp) internalFinalizeBlock(ctx context.Context, req *abci.Request
 
 	// GasMeter must be set after we get a context with updated consensus params.
 	gasMeter := app.getBlockGasMeter(app.finalizeBlockState.Context())
-	app.finalizeBlockState.SetContext(app.finalizeBlockState.Context().WithBlockGasMeter(gasMeter))
+	app.finalizeBlockState.SetContext(
+		app.finalizeBlockState.Context().
+			WithBlockGasMeter(gasMeter).
+			WithTxCount(len(req.Txs)),
```

Comment:
should WithTxCount when reset https://github.com/cosmos/cosmos-sdk/blob/e7bcfc23b4f0be09ae8a984fbcc125631453c66e/baseapp/abci.go#L794